test/support/test_utils.ex

```diff
@@ -419,26 +419,20 @@ defmodule Logflare.TestUtils do
   def random_pos_integer(limit \\ 1000), do: :rand.uniform(limit)
 
   @doc """
-  Fetches data until the asserts is able to run its asserts a certain amount of times (default 10 times).
+  Run function `times` times and will retry failed assertions
   """
-  @spec retry_fetch(
-          (-> :retry | any()),
-          (any() -> :retry | any()),
-          pos_integer(),
-          pos_integer()
-        ) ::
-          :ok | {:error, :max_retries_reached}
-  def retry_fetch(fetcher, asserts, times \\ 10, sleep \\ 50)
-  def retry_fetch(_, _, 0, _), do: {:error, :max_retries_reached}
-
-  def retry_fetch(fetcher, asserts, times, sleep) do
-    with fetcher_result when fetcher_result != :retry <- fetcher.(),
-         assert_result when assert_result != :retry <- asserts.(fetcher_result) do
-      send(self(), :done)
-    else
-      :retry ->
-        :timer.sleep(sleep)
-        retry_fetch(fetcher, asserts, times - 1)
+  @spec retry_assert(times :: pos_integer(), sleep :: pos_integer(), func :: (-> any())) :: any()
+  def retry_assert(times \\ 10, sleep \\ 50, func) do
```

Comment:
alternate implementation is to keep retrying with 50ms intervals for a set timeout, then caller can adjust the timeout with a similar api like `assert_receive`

Reply:
I can change the implementation, that shouldn't be too hard to implement.